### PR TITLE
Demo app polishing.

### PR DESCRIPTION
**Pull request by tjprescott**
Updates to the Swift demo app. Adds activity spinner and customizable pre-play buffering. Adds BlobDownloaderDelegate, as a better alternative to callback closures.

### CODE REVIEW

File: examples/AzureSDKDemoSwift/Source/BlobTableViewController.swift [hunk 1]
```diff
@@ -160,7 +160,7 @@ class BlobTableViewController: UITableViewController, MSALInteractiveDelegate {
                 showAlert(error: "Unable to create URL!")
                 return
             }
-            self.showActivitySpinner()
+            showActivitySpinner()
```
Comment (bsiegel) [inline examples/AzureSDKDemoSwift/Source/BlobTableViewController.swift]: It seems like swiftformat is not picking up the config file, these changes shouldn't be showing up. If you just run `swiftformat sdk` from the root of the repo, does it revert these changes?

Comment (bsiegel) [reply examples/AzureSDKDemoSwift/Source/BlobTableViewController.swift]: You can also explicitly pass the config file path with `--config <repo_root>/.swiftformat` but that shouldn't be necessary.

File: sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamDownloader.swift [hunk 2]
```diff
@@ -318,22 +329,41 @@ public class BlobStreamDownloader {
...
+            }
         }
     }
 
     /// Download the contents of this file to a stream.
     /// - Parameters:
     ///   - group: An optional `DispatchGroup` to wait for the download to complete.
-    public func next(inGroup group: DispatchGroup? = nil) {
+    ///   - completion: A completion handler with which to process the downloaded chunk.
+    public func next(inGroup group: DispatchGroup? = nil, then completion: @escaping (Result<Data, Error>, HTTPResponse) -> Void) {
```
Comment (bsiegel) [inline sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamDownloader.swift]: The type for `completion` should be `HTTPResultHandler<Data>` (the type alias just makes things simpler).

File: examples/AzureSDKDemoSwift/Source/BlobTableViewController.swift (hunk 1 above)
Comment (tjprescott) [reply examples/AzureSDKDemoSwift/Source/BlobTableViewController.swift]: I think swiftformat is working, because I ran a build and it made a bunch of changes I had to commit. This was likely swiftformat undoing parts that may have been reintroduced by a rebase.

File: sdk/appconfiguration/AzureAppConfiguration/Source/Models/ConfigurationSetting.swift [hunk 3]
```diff
@@ -54,7 +54,7 @@ public struct ConfigurationSetting: Codable {
         self.label = label
         self.tags = tags
         self.contentType = contentType
-        self.etag = eTag
+        etag = eTag
```
Comment (bsiegel) [inline sdk/appconfiguration/AzureAppConfiguration/Source/Models/ConfigurationSetting.swift]: Here's an example where swiftlint should not have made this change. Very odd. I will check out your branch and try running it locally.

Comment (bsiegel) [reply sdk/appconfiguration/AzureAppConfiguration/Source/Models/ConfigurationSetting.swift]: Just tried this and swiftformat undid a bunch of changes including this one.

Comment (tjprescott) [reply sdk/appconfiguration/AzureAppConfiguration/Source/Models/ConfigurationSetting.swift]: Hmm...yeah, I see what you mean. I just pushed that simple change you recommended about the alias and locally ran `swiftformat`. It made a bunch of changes. So it seems that its inclusion in the `swiftlint.sh` script isn't always working as intended.

File: sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamDownloader.swift (hunk 2 above)
Comment (tjprescott) [reply sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamDownloader.swift]: Fixed.
